test/server/controllers/transferwise.test.ts

```diff
@@ -79,33 +79,32 @@ describe('server/controllers/transferwise', () => {
     res.end = sandbox.stub().returns(res);
     res.setHeader = sandbox.stub().returns(res);
 
-    sandbox.stub(transferwise, 'getTemporaryQuote').resolves({
+    quote = {
       id: 1234,
       sourceCurrency: 'USD',
       targetCurrency: 'EUR',
       sourceAmount: 101.14,
       targetAmount: 90.44,
       rate: 0.9044,
       payOut: 'BANK_TRANSFER',
-      paymentOptions: [
-        {
-          formattedEstimatedDelivery: 'by March 18th',
-          estimatedDeliveryDelays: [],
-          allowedProfileTypes: ['PERSONAL', 'BUSINESS'],
-          payInProduct: 'BALANCE',
-          feePercentage: 0.0038,
-          estimatedDelivery: '2021-03-18T12:45:00Z',
-          fee: { transferwise: 3.79, payIn: 0, discount: 0, total: 3.79, priceSetId: 134, partner: 0 },
-          payIn: 'BALANCE',
-          sourceAmount: 101.14,
-          targetAmount: 90.44,
-          sourceCurrency: 'USD',
-          targetCurrency: 'EUR',
-          payOut: 'BANK_TRANSFER',
-          disabled: false,
-        },
-      ],
-    });
+      paymentOption: {
+        formattedEstimatedDelivery: 'by March 18th',
+        estimatedDeliveryDelays: [],
+        allowedProfileTypes: ['PERSONAL', 'BUSINESS'],
+        payInProduct: 'BALANCE',
+        feePercentage: 0.0038,
+        estimatedDelivery: '2021-03-18T12:45:00Z',
+        fee: { transferwise: 3.79, payIn: 0, discount: 0, total: 3.79, priceSetId: 134, partner: 0 },
+        payIn: 'BALANCE',
+        sourceAmount: 101.14,
+        targetAmount: 90.44,
+        sourceCurrency: 'USD',
+        targetCurrency: 'EUR',
+        payOut: 'BANK_TRANSFER',
+        disabled: false,
+      },
+    };
+    sandbox.stub(transferwise, 'quoteExpense').resolves(quote);
```

Comment:
Are we sure that wise will never return `paymentOptions` (array) in any of the requests? If not, should we keep supporting for both, e.g. with a monkey patch like:

```ts
if (!quote.paymentOption && quote.paymentOptions?.length) {
  quote.paymentOption = quote.paymentOptions[0];
}
```

Also, did they announce the breaking change somewhere?

Reply:
@Betree, Wise will return an array from the API but the `quoteExpense` is an internal function in the payment providers' libs that will return a quote with the selected BALANCE option already selected and inserted in `quote.paymentOption`.

In retrospect, I don't think it was such a great idea to mutate the original quote to persist this information, but it is how it currently works.

Reply:
(Everything is properly typed tho, so it shouldn't be insecure)